### PR TITLE
Bump gl-native to v10.6.0, common to v22.0.0.

### PR DESCRIPTION
**Pull request by pengdev**
## Features ✨ and improvements 🏁
* Update mapbox-common to v22.0.0.
* Reduce geometry on globe tile to increase rendering performance.
* Minimise tiles relayout on camera zooming with globe view. Improve the overall performance and reduce latency of the cached tiles appearance.
* Add minimum and maximum range check for sky layer property 'sky-atmosphere-sun' and 'sky-gradient-center'. If the input is invalid, the default property value will be used.
* Add support for sending turnstile event in native for embedded platforms.
* Use a single shared buffer across all globe tiles to increase performance.
* Use shaders to calculate the 'line-trim-offset' property update.
* Re-introduce partial tile loading to provide better TTRC.
* The TilesetDescriptorOptions::pixelRatio parameter is now passed to the Tile Store and considered for the raster tile pack loading

## Bug fixes 🐞
* Fix lag during night/day style switching when globe is enabled.
* Fix crash on globe symbol placement that happens during style change.
* Fix incorrect image source rendering with terrain enabled.
* Fix in-memory tile cache for non-geometry tiles when the map projection changes. Before, the tiles cached while the previous projection was active could not be used with the new projection active.
* Fix unexpected request delay for tiles taken from in-memory cache when minimumTileUpdateInterval is set.
* Fix 'line-trim-offset' input losing precision via shader calculation.
* Fix momentary appearance of pan tiles labels.
* Fix for location indicator not rendered at horizon when terrain is enabled.
* Fix for 'line-trim-offset' calculation not properly covering 'round' or 'square' line cap in line ends.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,12 +17,57 @@ Mapbox welcomes participation and contributions from everyone.
...
+* Fix incorrect image source rendering with terrain enabled. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix in-memory tile cache for non-geometry tiles when the map projection changes. Before, the tiles cached while the previous projection was active could not be used with the new projection active. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix unexpected request delay for tiles taken from in-memory cache when minimumTileUpdateInterval is set. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
 
 ## Dependencies
+* Bump Mapbox Android base library to v0.8.0. ([#1323](https://github.com/mapbox/mapbox-maps-android/pull/1323))
+* Bump gl-native to v10.6.0, common to 22.0.0. ([#1351](https://github.com/mapbox/mapbox-maps-android/pull/1351), [#1354](https://github.com/mapbox/mapbox-maps-android/pull/1354), [#1359](https://github.com/mapbox/mapbox-maps-android/pull/1359), [#1396](https://github.com/mapbox/mapbox-maps-android/pull/1396), [#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
```
Comment (yunikkk) [inline CHANGELOG.md:70]: Do we need the whole list with beta-rc PRs? Isn't just #1432 enough?

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,12 +17,57 @@ Mapbox welcomes participation and contributions from everyone.
...
 * Fix the delay in the first appearance of the location puck. ([1403](https://github.com/mapbox/mapbox-maps-android/pull/1403))
 * Remove `android.permission.GET_TASKS` permission from the SDK. ([1430](https://github.com/mapbox/mapbox-maps-android/pull/1430))
+* Fix lag during night/day style switching when globe is enabled. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix crash on globe symbol placement that happens during style change. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix incorrect image source rendering with terrain enabled. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix in-memory tile cache for non-geometry tiles when the map projection changes. Before, the tiles cached while the previous projection was active could not be used with the new projection active. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
+* Fix unexpected request delay for tiles taken from in-memory cache when minimumTileUpdateInterval is set. ([#1432](https://github.com/mapbox/mapbox-maps-android/pull/1432))
```
Comment (yunikkk) [inline CHANGELOG.md:66]: This line is added at https://github.com/mapbox/mapbox-maps-android/pull/1432/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR51

Comment (yunikkk) [reply CHANGELOG.md:66]: In the gl-native changelog it actually points to the single PR 3516

Comment (pengdev) [reply CHANGELOG.md]: Removed

File: CHANGELOG.md (hunk 1 above)
Comment (pengdev) [reply CHANGELOG.md]: yeah, agree, updated.

**Review by yunikkk — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Pull request merged**